File: Easy/Python/605. Can Place Flowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        index = 0

        while index < len(flowerbed):
            if flowerbed[index] == 0 and (index == 0 or flowerbed[index - 1] == 0) and\
               (index == len(flowerbed) - 1 or flowerbed[index + 1] == 0):
                flowerbed[index] = 1
                n -= 1
            if n <= 0:
                return True
            index += 1

        return False

File: Easy/Java/605. Can Place Flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int index = 0;

        while(index < flowerbed.length){
            if(flowerbed[index] == 0 && (index == 0 || flowerbed[index - 1] == 0) &&
              (index == flowerbed.length - 1 || flowerbed[index + 1] == 0)){
                flowerbed[index] = 1;
                n--;
            }
            if(n <= 0)
                return true;
            index++;
        }

        return false;
    }
}
